cpp/turbodbc_arrow/Test/tests/arrow_result_set_test.cpp

```diff
@@ -0,0 +1,436 @@
+#include <turbodbc_arrow/arrow_result_set.h>
+
+#undef BOOL
+#include <arrow/api.h>
+#include <arrow/test-util.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <sql.h>
+
+
+struct mock_result_set : public turbodbc::result_sets::result_set
+{
+	MOCK_METHOD0(do_fetch_next_batch, std::size_t());
+	MOCK_CONST_METHOD0(do_get_column_info, std::vector<turbodbc::column_info>());
+	MOCK_CONST_METHOD0(do_get_buffers, std::vector<std::reference_wrapper<cpp_odbc::multi_value_buffer const>>());
+};
+
+
+TEST(ArrowResultSetTest, SimpleSchemaConversion)
+{
+	mock_result_set rs;
+	std::vector<turbodbc::column_info> expected = {{
+    "int_column", turbodbc::type_code::integer, true}};
+	EXPECT_CALL(rs, do_get_column_info()).WillRepeatedly(testing::Return(expected));
+
+  turbodbc_arrow::arrow_result_set ars(rs);
+  auto schema = ars.schema();
+  ASSERT_EQ(schema->num_fields(), 1);
+  auto field = schema->field(0);
+  ASSERT_EQ(field->name(), "int_column");
+  ASSERT_EQ(field->type(), arrow::int64());
+  ASSERT_EQ(field->nullable(), true);
+}
+
+TEST(ArrowResultSetTest, AllTypesSchemaConversion)
```

Comment:
I'll refactor some of the tests but this one is quite comprehensive already. Splitting it up will probably not produce any better understandable code.